src/rules/add-roles-from-sqlserver.js

```diff
@@ -11,7 +11,7 @@
  */
 
 function addRolesFromSqlServer(user, context, callback) {
-  const tedious = require('tedious');
+  const tedious = require('tedious@6.6.2');
```

Comment:
What are the long-term implications of pinning the version in the rules? It's probably minimal if we assume we will never change the default version of the package in `webtask-modules`, in which case the tenant always has to pin a version to use anything other than the default.

If the default version of `tedious` package no longer works prior to this change, however, should we instead change the default version in `webtask-modules`? This is likely the case per the ESD report. It means any tenant using the default version is already broken.

Reply:
This is a complicated one, and a special case for tedious.

The root cause of the issue here is that Node 12 is resolving a different default version of tedious than Node 8 by default. The Webtask team's explanation about why is related to [this commit](https://github.com/auth0/webtask-monorepo/commit/91e708933b4506d7bd65f146c334b891286a1da0). 

Webtask uses the first module lists in its package set to determine what package is loaded by default in rules. However, in the commit linked above, `tedious@6.6.2` was added to the _beginning_ of the package list. Since Node 8's base package volume was already built, it had no effect on changing the default. However, when we built Node 12, the volume was built from scracth, and so `tedious@6.6.2` is now the  default in Node 12 while `1.11.0` the default in Node 8.

Rather than forking the template scripts by Node version, I included the version here so that the same template should work in either Node 8 or Node 12 (for people who are making new rules based on this template). But, to your point: any tenant using a default version of tedious in their rules will probably break when they from Node 8 to Node 12 because the version is changing out from underneath them (and this template change won't help them).

I wonder if we should add some documentation about this specific package to the Node 12 migration guide? Thoughts?

Reply:
But aren't they already broken when on Node 8 with `tedious@1.11.0`?

Would be better not to have an exception if possible.

Reply:
The existing template works on Node 8 because `tedious@1.11.0` is the default there. That same code does not work in Node 12 where `tedious@6.6.2` is the default.

If we do a clean build of Webtask for Node 8, then any existing rules for tenants on Node 8 that use tedious will break because we would silently upgrade them to `tedious@6.6.2`.

One other possibility is to see if we have any Node 12 tenants using a rule that includes tedious. If not, we would be free to change the default version in Node 12 back to `tedious@1.11.0` so it matches Node 8.